### PR TITLE
Replace exceptions by errors

### PR DESCRIPTION
**Pull request by saraycp**
Set errors instead of exceptions in WorkflowStepsValidator and WorkflowFiltersValidator, so the user can know all the non-valid attributes at once.

Also, improve error messages and adapt specs.

### CODE REVIEW

File: src/api/app/validators/workflow_steps_validator.rb [hunk 1]
```diff
@@ -30,14 +29,20 @@ def no_steps?
 
   def errors
     acc = []
```
Comment (vpereira) [inline src/api/app/validators/workflow_steps_validator.rb:31]: what is `acc`? couldn't we use a better name here?

Comment (saraycp) [reply src/api/app/validators/workflow_steps_validator.rb:31]: definitely!

Comment (saraycp) [reply src/api/app/validators/workflow_steps_validator.rb]: done

**Comment by saraycp** [comment]
@vpereira I've added an extra test in a new commit.

**Review by vpereira — APPROVED** [review]

**Comment by saraycp** [comment]
> Errors `UnsupportedWorkflowFilterTypes` and `InvalidWorkflowStepDefinition` are no longer raised. We could remove their definitions, in `app/models/token/errors.rb` and in `app/models/workflow/errors.rb`.

Thanks! Updated @eduardoj

**Review by eduardoj — APPROVED** [review]

**Pull request merged**